### PR TITLE
VertX 4.5.x instrumentation

### LINKED ISSUE #1618: Support for Vert.x v4.5.0
Vert.x instrumentation fails to apply starting at v4.5.0.

```
WeaveViolation{type=MISSING_ORIGINAL_BYTECODE, clazz=io/vertx/core/impl/ContextBase}
```

https://github.com/vert-x3/wiki/wiki/4.5.0-Release-Notes

### PR DESCRIPTION
**Pull request by meiao**
### Overview
Adds instrumentation for VertX 4.5 web client and futures.

### Related Github Issue
Fixes #1618 

### Checks

- [x] Your contributions are backwards compatible with relevant frameworks and APIs.
- [x] Your code does not contain any breaking changes. Otherwise please describe. 
- [x] Your code does not introduce any new dependencies. Otherwise please describe.

### CODE REVIEW

File: instrumentation/vertx-core-3.3.0/src/main/java/com/nr/vertx/instrumentation/VertxCoreUtil.java [hunk 1]
```diff
@@ -56,6 +56,7 @@ public static void processResponse(Segment segment, HttpClientResponseImpl resp,
                                                    .uri(uri)
                                                    .procedure(END)
                                                    .inboundHeaders(new InboundWrapper(resp))
+                                                   .status(resp.statusCode(), resp.statusMessage())
```
Comment (meiao) [inline instrumentation/vertx-core-3.3.0/src/main/java/com/nr/vertx/instrumentation/VertxCoreUtil.java:59]: Changes in modules < 4.5.0 are mostly adding the status code and message to incoming responses.

File: instrumentation/vertx-core-4.0.0/src/test/java/com/nr/vertx/instrumentation/VertxClientTest.java [hunk 2]
```diff
@@ -55,10 +55,8 @@ public static void beforeClass() {
         server = vertx.createHttpServer().requestHandler(request -> {
             final String statusCode = request.getHeader("statusCode");
             if (statusCode == null) {
-                System.out.println("statusCode is null");
```
Comment (meiao) [inline instrumentation/vertx-core-4.0.0/src/test/java/com/nr/vertx/instrumentation/VertxClientTest.java:58]: Also removed a bunch of System.outs. Mostly from tests.

File: instrumentation/vertx-core-4.5.0/build.gradle [hunk 3]
```diff
@@ -0,0 +1,22 @@
+dependencies {
```
Comment (meiao) [inline instrumentation/vertx-core-4.5.0/build.gradle:1]: This module is basically a copy of the 4.3.2 module.
The difference is a class that was renamed, and implementing a new method that was added to an interface.

File: instrumentation/vertx-core-4.5.1/build.gradle [hunk 4]
```diff
@@ -0,0 +1,19 @@
+dependencies {
```
Comment (meiao) [inline instrumentation/vertx-core-4.5.1/build.gradle:1]: And this module is a copy of 4.5.0 but with a method with visibility changed.

File: settings.gradle [hunk 5]
```diff
@@ -363,6 +363,25 @@ include 'instrumentation:tomcat-10'
...
+include 'instrumentation:vertx-core-3.3.3'
+include 'instrumentation:vertx-core-3.4.1'
+include 'instrumentation:vertx-core-3.6.0'
+include 'instrumentation:vertx-core-3.8.0'
+include 'instrumentation:vertx-core-3.9.0'
+include 'instrumentation:vertx-core-4.0.0'
+include 'instrumentation:vertx-core-4.1.0'
+include 'instrumentation:vertx-core-4.3.2'
+include 'instrumentation:vertx-core-4.5.0'
+include 'instrumentation:vertx-core-4.5.1'
```
Comment (meiao) [inline settings.gradle:376]: I alphabetized all the vertx modules, and added the 4.5.0 and 4.5.1.

**Review by obenkenobi — APPROVED** [review]

**Pull request merged**